File: backend/app/schemas/pathway.py
from pydantic import BaseModel


class PathwayBlockSchema(BaseModel):
    id: str
    block_type: str
    category: str
    label: str
    config: dict
    position: dict | None = None
    order_index: int = 0


class PathwayEdgeSchema(BaseModel):
    id: str
    source_block_id: str
    target_block_id: str
    edge_type: str = "default"
    label: str | None = None


class PathwayListItem(BaseModel):
    id: str
    name: str
    description: str | None
    condition: str | None
    target_tiers: list[int]
    status: str
    version: int
    block_count: int = 0
    created_at: str
    updated_at: str


class PathwayDetail(PathwayListItem):
    created_by: str
    published_at: str | None
    published_by: str | None
    blocks: list[PathwayBlockSchema]
    edges: list[PathwayEdgeSchema]


class PathwayCreate(BaseModel):
    name: str
    description: str | None = None
    condition: str | None = None
    target_tiers: list[int] = []


class PathwayUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    condition: str | None = None
    target_tiers: list[int] | None = None
    status: str | None = None


class BlockCreate(BaseModel):
    block_type: str
    category: str
    label: str
    config: dict = {}
    position: dict | None = None
    order_index: int = 0


class BlockUpdate(BaseModel):
    label: str | None = None
    config: dict | None = None
    position: dict | None = None
    order_index: int | None = None


class PathwayListResponse(BaseModel):
    items: list[PathwayListItem]
    total: int


class PathwayGenerateRequest(BaseModel):
    prompt: str
    pathway_id: str | None = None


class AISessionListItem(BaseModel):
    id: str
    title: str
    pathway_id: str | None
    message_count: int
    created_at: str
    updated_at: str


class AISessionDetail(AISessionListItem):
    messages: list[dict]
    generated_pathway: dict | None


class AISessionCreate(BaseModel):
    title: str = "New Chat"


class AISessionUpdate(BaseModel):
    title: str | None = None
    messages: list[dict] | None = None
    generated_pathway: dict | None = None
    pathway_id: str | None = None

File: src/services/types/pathway.ts
export interface PathwayBlockSchema {
  id: string;
  block_type: string;
  category: string;
  label: string;
  config: Record<string, unknown>;
  position: { x: number; y: number } | null;
  order_index: number;
}

export interface PathwayEdgeSchema {
  id: string;
  source_block_id: string;
  target_block_id: string;
  edge_type: string;
  label: string | null;
}

export interface PathwayListItem {
  id: string;
  name: string;
  description: string | null;
  condition: string | null;
  target_tiers: number[];
  status: string;
  version: number;
  block_count: number;
  created_at: string;
  updated_at: string;
}

export interface PathwayDetail extends PathwayListItem {
  created_by: string;
  published_at: string | null;
  published_by: string | null;
  blocks: PathwayBlockSchema[];
  edges: PathwayEdgeSchema[];
}

export interface PathwayCreate {
  name: string;
  description?: string;
  condition?: string;
  target_tiers?: number[];
}

export interface PathwayUpdate {
  name?: string;
  description?: string;
  condition?: string;
  target_tiers?: number[];
  status?: string;
}

export interface BlockCreate {
  block_type: string;
  category: string;
  label: string;
  config?: Record<string, unknown>;
  position?: { x: number; y: number };
  order_index?: number;
}

export interface BlockUpdate {
  label?: string;
  config?: Record<string, unknown>;
  position?: { x: number; y: number };
  order_index?: number;
}

export interface PathwayListResponse {
  items: PathwayListItem[];
  total: number;
}

export interface PathwayGenerateRequest {
  prompt: string;
  pathway_id?: string;
}

export interface AIGeneratedBlock {
  block_type: string;
  category: string;
  label: string;
  config: Record<string, unknown>;
  order_index: number;
}

export interface AIGeneratedEdge {
  source_index: number;
  target_index: number;
  edge_type: string;
  label?: string;
}

export interface AIGeneratedPathway {
  name: string;
  description: string;
  condition: string;
  target_tiers: number[];
  blocks: AIGeneratedBlock[];
  edges: AIGeneratedEdge[];
}

export interface PathwayGenerateResponse {
  message: string;
  is_complete: boolean;
  pathway: AIGeneratedPathway;
}

// ── AI Sessions ─────────────────────────────────────────────────────────

export interface AISessionListItem {
  id: string;
  title: string;
  pathway_id: string | null;
  message_count: number;
  created_at: string;
  updated_at: string;
}

export interface AISessionDetail extends AISessionListItem {
  messages: Array<{ role: string; content: string }>;
  generated_pathway: AIGeneratedPathway | null;
}

export interface AISessionCreate {
  title?: string;
}

export interface AISessionUpdate {
  title?: string;
  messages?: Array<{ role: string; content: string }>;
  generated_pathway?: AIGeneratedPathway | null;
  pathway_id?: string;
}
